Makefile

```diff
@@ -574,12 +574,21 @@ misspell:
 yamllint:
 	hack/check/check-yaml.sh
 
+
+# These are the modules that still contain issues reported by golangci-lint.
+# Once a module is updated to be lint-free, remove from list to catch lint regressions.
+MODULES_NEEDING_LINT_FIX=./cmd/cli/plugin/login ./cmd/cli/plugin/package ./cmd/cli/plugin-admin/builder . ./apis/core ./apis/addonconfigs ./capabilities/client ./addons ./hack/packages/package-tools ./packageclients ./tkg ./tkr ./pinniped-components/post-deploy ./pkg/v1/tkr
+
 go-lint: tools ## Run linting of go source
 	@for i in $(GO_MODULES); do \
 		echo "-- Linting $$i --"; \
-		pushd $${i}; \
-		$(GOLANGCI_LINT) run -v --timeout=10m; \
-		popd; \
+		pushd $${i} > /dev/null; \
+	    if echo ${MODULES_NEEDING_LINT_FIX} | tr ' ' '\n' | grep -q -x $$i; then \
```

Comment:
Nit: replace spaces with tab to match indentation

Nice use of `-x` to handle the lone `.` path.  That's now a new grep flag in my arsenal 👍

Reply:
updated